backend/app/user_handler.go

```diff
@@ -0,0 +1,325 @@
+	if request.Password != request.ConfirmPassword {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "password and confirm password don't match"})
+		return
+	}
+
+	// check if user previously exists
+	_, err := h.db.GetUserByEmail(request.Email)
+	if err == nil {
+		c.JSON(http.StatusConflict, gin.H{"error": "user already registered"})
+		return
+	}
+
+	// hash password
+	salt, err := internal.GenerateSalt()
+	if err != nil {
+		log.Error().Err(err).Msg("Error Generating salt for password")
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "error hashing password"})
+		return
+	}
+	hashed := internal.HashPassword(request.Password, salt)
+
+	isAdmin := internal.IsAdmin(request.Email, h.config.Admins)
+
+	user := models.User{
+		Username: request.Name,
+		Email:    request.Email,
+		Password: hashed,
+		Salt:     salt,
+		Admin:    isAdmin,
+	}
+
+	// create user model in db
+	err = h.db.RegisterUser(&user)
+	if err != nil {
+		log.Error().Err(err).Send()
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to register user"})
+		return
+	}
+
+	saved, _ := h.db.GetUserByEmail(user.Email)
+
+	// create token pairs
+	tokenPair, err := h.tokenManager.CreateTokenPair(saved.ID, saved.Username, isAdmin)
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to generate token pair")
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to generate tokens"})
+		return
+	}
+	c.JSON(http.StatusCreated, tokenPair)
+
+}
+
+// LoginUserHandler logs user into the system
+func (h *Handler) LoginUserHandler(c *gin.Context) {
+	var request LoginInput
+
+	// check on request format
+	if err := c.ShouldBindJSON(&request); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request format"})
+		return
+	}
+
+	// get user by email
+	user, err := h.db.GetUserByEmail(request.Email)
+	if err != nil {
+		log.Error().Err(err).Msg("failed to get user by email")
+		c.JSON(http.StatusBadRequest, gin.H{"error": "email or password is incorrect"})
+		return
+
+	}
+
+	// verify password
+	match := internal.VerifyPassword(user.Password, request.Password, user.Salt)
+	if !match {
+		c.JSON(http.StatusUnauthorized, gin.H{"error": "email or password is incorrect"})
+		return
+	}
+
+	// create token pairs
+	tokenPair, err := h.tokenManager.CreateTokenPair(user.ID, user.Username, user.Admin)
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to generate token pair")
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to generate tokens"})
+		return
+	}
+	c.JSON(http.StatusCreated, tokenPair)
+
+}
+
+// RefreshTokenHandler handles token refresh requests
+func (h *Handler) RefreshTokenHandler(c *gin.Context) {
+	var request RefreshTokenInput
+
+	if err := c.ShouldBindJSON(&request); err != nil {
+		log.Error().Err(err).Send()
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request format"})
+		return
+	}
+
+	accessToken, err := h.tokenManager.RefreshAccessToken(request.RefreshToken)
```

Comment:
misleading names
- RefreshAccessToken function but you gave it a refresh token then output is access token?